### PR TITLE
Fixes no visible binding for global variable for PEcAn.benchmark package | Fixes #2758

### LINKED ISSUE #2758: Fix tidyverse "no visible binding for global variable" notes
Adapted from discussion with @infotroph in https://github.com/PecanProject/pecan/pull/2756

- an easy but tedious and inelegant fix for a problem that's mostly just annoying
- We are currently ignoring 571 `no visible binding for global variable` NOTEs across all of PEcAn, and my wild guess is that half to two-thirds of these are Tidyverse calls.

**Shortcu:t** For anyone who runs into this error and doesn't want to implement this fix: just add the appropriate note causing the build to fail to the package's `<pkg>/test/Rcheck_reference.log`, like this: https://github.com/PecanProject/pecan/pull/2756/commits/4a303b7fe3b32914c1f329be5efffc5f55030339

**If you are in a good place to tackle this:** we support you, get comfy! Bonus points if you fix all of the issues in the package you are working on. A **special prize** if you fix all (currently 563) of these Notes.

TODO:

* [ ] find list of violations in `<pkg>/test/Rcheck_reference.log` files
  * ` find . -name Rcheck_reference.log | xargs grep "no visible binding for global variable"` 
  *  to count occurrences add ` | wc -l` to the end of the line above
* [ ] prepend variables with `.data$` as done here: https://github.com/PecanProject/pecan/pull/2756/files#diff-296112e719b8455a8be63e9cac941111d4ae1700081aa05b3a9adfa8d5b9738aR210
* [ ] add rlang to Imports for relevant package to description
* [ ] add `##' @importFrom rlang .data` to `<pkg>R/PEcAn.<pkgname>-package.R`. See [PEcAn.DB](https://github.com/PecanProject/pecan/blob/develop/base/db/R/PEcAn.DB-package.R#L8) for an example.
  * can also be done per function, but not sure when this would be preferred
* [ ] delete (relevant) lines starting `<function name>: no visible binding for global variable <variable>` in `<pkg>/test/Rcheck_reference.log`
* [ ] update documentation

Alternative solutions:
- change build checks to allow these notes
- or add notes as they appear to `<pkg>/test/Rcheck_reference.log` so that build ignores it
- wait until a more elegant solution evolves

See also 
- https://community.rstudio.com/t/how-to-solve-no-visible-binding-for-global-variable-note/28887
- https://www.r-bloggers.com/2019/08/no-visible-binding-for-global-variable/
- https://www.google.com/search?q=no+visible+binding+for+global+variable
- https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html

_Originally posted by @infotroph in https://github.com/PecanProject/pecan/pull/2756#discussion_r544964692_

### PR DESCRIPTION
**Pull request by Sarthakaga15**
@infotroph @ashiklom @moki1202 I tried fixing a few no visible binding errors; I would love your opinion on proceeding further. Please let me know if my changes are wrong so I can work on fixing it.

### CODE REVIEW

**Review by ashiklom — APPROVED** [review]
This looks great to me!

Minor request: Could you also remove the corresponding lines from the [Rcheck_reference.log](https://github.com/PecanProject/pecan/blob/develop/modules/benchmark/tests/Rcheck_reference.log#L68)? That help us catch this problem if it resurfaces in the future.

Thanks for your help!

**Comment by Sarthakaga15** [comment]
@ashiklom Apologies . I forgot to convert this into the draft. I will be updating this PR further, fixing some more global binding errors ;)

**Comment by infotroph** [comment]
@Sarthakaga15 To make the CI checks pass you'll eventually also need to make sure that `rlang::.data` is being imported into all the packages you use it in, via `@importFrom rlang .data`. Feel free to ping me if you need more details.

File: modules/benchmark/R/align_data_to_data_pft.R [hunk 1]
```diff
@@ -50,6 +50,7 @@
 #'  custom_table = table)
 #' }
 #' @export
+#' @importFrom rlang .data
```
Comment (infotroph) [inline modules/benchmark/R/align_data_to_data_pft.R:53]: I don't see anywhere in this function that uses .data, so unless I missed one I don't think you need this `importFrom`

**Review by infotroph — APPROVED** [review]

File: modules/benchmark/DESCRIPTION [hunk 2]
```diff
@@ -35,6 +35,8 @@ Imports:
     SimilarityMeasures,
```
Comment (infotroph) [inline modules/benchmark/DESCRIPTION:35]:
```suggestion
    rlang,
    SimilarityMeasures,
```

**Pull request merged**